### PR TITLE
refactor(cardano-services): prepare for making all services a HTTP microservice for monitoring

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
- All services need to expose an HTTP server for monitoring, so our current name, for what is the provider server, is too generic.
- `HttpService` is currently [loosely](https://github.com/input-output-hk/cardano-js-sdk/blob/0d52db72cf548c5072df4ad7562258cb8c2118a3/packages/core/src/Provider/Provider.ts#L24) bound to the concept of `Provider`
- The current service factory could be utilised by new services, rather than creating new ones.
- We're missing the desired `/ready` endpoint for HTTP status code-based monitoring for orchestration.

# Proposed Solution
- Rename the existing `http-server` service to `provider-server`
- Since we're not wanting to make large refactors, I've elected to stick with the dependency of a Provider in the HttpService, as it can easy be adhered with another service, that contains a `healthCheck` method, by casting.
- Add the ready endpoint that simply interprets the healthcheck and responds with an appropriate HTTP status code..

# Important Changes Introduced
- Refactoring the existing RabbitMQ worker is out of scope due to the more complicated migration needed. Will be done when it's needed:  `ADP-2782`

### CODE REVIEW

**Review by iccicci — DISMISSED** [review]
Great job!

**Review by mkazlauskas — CHANGES_REQUESTED** [review]
Looking good, but the tests are failing

File: packages/cardano-services/src/Http/HttpServer.ts [hunk 1]
```diff
@@ -119,8 +119,20 @@ export class HttpServer extends RunnableModule {
       }
       return HttpServer.sendJSON(res, body);
     };
+    const servicesReadyCheckHandler = async (req: express.Request, res: express.Response) => {
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply packages/cardano-services/src/Http/HttpServer.ts]: https://github.com/input-output-hk/cardano-js-sdk/pull/622/commits/aca2c40edc3aec2120969c5ca680763c1d8639cf

**Review by mkazlauskas — DISMISSED** [review]
Nice work! :rocket:

**Review by iccicci — DISMISSED** [review]

**Review by iccicci — APPROVED** [review]
Now I used this, I understood how much powerful it is 🚀

**Review by mkazlauskas — DISMISSED** [review]

**Review by iccicci — APPROVED** [review]

**Pull request merged**